### PR TITLE
Bugfix - removing the code to take backup of keys. 

### PR DESCRIPTION
**Pull request by dpksinghal**
This causes issues in case of multiple agents/multiple configurations on the machine.

### CODE REVIEW

File: src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs [hunk 1]
```diff
@@ -197,24 +197,12 @@ private void RevertAutoLogonSpecificSettings(string domainName, string userName)
...
-                RevertOriginalValue(RegistryHive.LocalMachine, 
-                                        RegistryConstants.MachineSettings.SubKeys.AutoLogon, 
-                                        RegistryConstants.MachineSettings.ValueNames.AutoLogon);
-                RevertOriginalValue(RegistryHive.LocalMachine,
-                                        RegistryConstants.MachineSettings.SubKeys.AutoLogon,
-                                        RegistryConstants.MachineSettings.ValueNames.AutoLogonUserName);
-                RevertOriginalValue(RegistryHive.LocalMachine,
-                                        RegistryConstants.MachineSettings.SubKeys.AutoLogon,
-                                        RegistryConstants.MachineSettings.ValueNames.AutoLogonDomainName);
-                RevertOriginalValue(RegistryHive.LocalMachine,
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs]: why we don't need `AutoLogonPassword` anymore?

File: src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs [hunk 2]
```diff
@@ -197,24 +197,12 @@ private void RevertAutoLogonSpecificSettings(string domainName, string userName)
...
-                                        RegistryConstants.MachineSettings.SubKeys.AutoLogon,
-                                        RegistryConstants.MachineSettings.ValueNames.AutoLogonPassword);
-                RevertOriginalValue(RegistryHive.LocalMachine, 
-                                        RegistryConstants.MachineSettings.SubKeys.AutoLogon,
-                                        RegistryConstants.MachineSettings.ValueNames.AutoLogonCount);
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs]: why we don't need `AutoLogonCount` anymore?

File: src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs (hunk 1 above)
Comment (dpksinghal) [inline src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs]: earlier the model was to take backup of existing settings during configuration of the agent.. I think it is over-kill and is causing issues in many scenarios (some of them I mentioned in the mail). now as part of unconfig I am just resetting the changes that were done during configuration.. AutoLogonPassword key doesn't play any role when AutoLogon is not enabled..

File: src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs (hunk 2 above)
Comment (dpksinghal) [inline src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs]: when autologon is not enabled these settings are not meaningful.. as part of unconfig we are just disabling the autologon hence these keys can be removed..

File: src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs (hunk 1 above)
Comment (ericsciple) [inline src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs]: What are the scenarios? Also I'm confused whether taking the backup is the problem or whether reverting during unconfigure is the problem.

**Comment by nigurr** [comment]
@dpksinghal If any test cases needs to be added/updated, please do them

**Comment by dpksinghal** [comment]
Have updated the existing L0 tests for these changes.

**Pull request merged**